src/internal/client.ts

```diff
@@ -740,4 +743,63 @@ export class TypedClient {
       (err) => cb(err),
     )
   }
+
+  /**
+   * Get part-info of all parts of an incomplete upload specified by uploadId.
+   */
+  protected async listParts(bucketName: string, objectName: string, uploadId: string): Promise<UploadedPart[]> {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError('Invalid bucket name: ' + bucketName)
+    }
+    if (!isValidObjectName(objectName)) {
+      throw new errors.InvalidObjectNameError(`Invalid object name: ${objectName}`)
+    }
+    if (!isString(uploadId)) {
+      throw new TypeError('uploadId should be of type "string"')
+    }
+    if (!uploadId) {
+      throw new errors.InvalidArgumentError('uploadId cannot be empty')
+    }
+
+    const parts: UploadedPart[] = []
+    let marker = 0
+    let result
```

Comment:
This should be typed, again

Reply:
this variable is set right 2 line below, personally I don't think it's necessary

Reply:
Okay